build.sbt

```diff
@@ -1,120 +1,110 @@
+    libraryDependencies ++= Dependencies.conseilApiInclude,
+    coverageExcludedPackages := Seq(
+          "<empty>",
+          ".*\\.Conseil",
+          ".*\\.ConseilAppConfig"
+        ).mkString(";")
+  )
+  .settings(Defaults.itSettings)
+  .configs(IntegrationTest)
+  .addRunCommand(
+    description = "A run application Task.",
+    javaExtras = Seq("-Xms1024M", "-Xmx8192M", "-Xss1M", "-XX:+CMSClassUnloadingEnabled")
+  )
+  .enableAssembly()
+  .dependsOn(common % "compile->test")
 
-scalacOptions ++= ScalacOptions.common
-scalacOptions += "-P:silencer:pathFilters=src/main/scala/tech/cryptonomic/conseil/tezos/Tables.scala"
-import complete.DefaultParsers._
+lazy val lorre = (project in file("conseil-lorre"))
+  .settings(
+    name := "conseil-lorre",
+    mainClass := Some("tech.cryptonomic.conseil.lorre.Lorre"),
+    libraryDependencies ++= Dependencies.conseilLorreInclude,
+    coverageExcludedPackages := Seq(
+          "<empty>",
+          ".*\\.Lorre",
+          ".*\\.LorreAppConfig"
+        ).mkString(";")
+  )
+  .settings(Defaults.itSettings)
+  .configs(IntegrationTest)
+  .addRunCommand(
+    description = "A run application Task.",
+    javaExtras = Seq("-Xmx512M", "-Xss1M", "-XX:+CMSClassUnloadingEnabled")
+  )
+  .enableAssembly()
+  .dependsOn(common)
 
-lazy val runConseil = inputKey[Unit]("A conseil run task.")
-fork in runConseil := true
-javaOptions in runConseil ++= Seq("-Xms1024M", "-Xmx8192M", "-Xss1M", "-XX:+CMSClassUnloadingEnabled")
-runConseil := Def.inputTaskDyn {
-  val args = spaceDelimited("").parsed
-  runInputTask(Runtime, "tech.cryptonomic.conseil.Conseil", args: _*).toTask("")
-}.evaluated
+lazy val lorreSchema = (project in file("conseil-lorre-schema"))
+  .settings(
+    name := "conseil-lorre-schema",
+    mainClass := Some("tech.cryptonomic.conseil.lorre.schema.GenSchema")
+  )
+  .addRunCommand(description = "A run schema generating Task.")
+  .enableAssembly()
+  .dependsOn(common)
 
-lazy val runLorre = inputKey[Unit]("A lorre run task.")
-fork in runLorre := true
-javaOptions ++= Seq("-Xmx512M", "-Xss1M", "-XX:+CMSClassUnloadingEnabled")
-runLorre := Def.inputTaskDyn {
-  val args = spaceDelimited("").parsed
-  runInputTask(Runtime, "tech.cryptonomic.conseil.Lorre", args: _*).toTask("")
-}.evaluated
+lazy val smokeTests = (project in file("conseil-smoke-tests"))
+  .settings(
+    name := "conseil-smoke-tests",
+    mainClass := Some("tech.cryptonomic.conseil.smoke.tests.RegressionRun"),
+    libraryDependencies ++= Dependencies.conseilSmokeTestsInclude,
+    coverageExcludedPackages := Seq(
+          "<empty>",
+          ".*\\.EndpointClient",
+          ".*\\.RegressionRun"
+        ).mkString(";")
+  )
+  .addRunCommand(description = "A run smoke tests Task.")
+  .disableAssembly()
 
-lazy val genSchema = taskKey[Unit]("A schema generating task.")
-fullRunTask(genSchema, Runtime, "tech.cryptonomic.conseil.scripts.GenSchema")
+addCommandAlias("runApi", "; api/runTask")
```

Comment:
`runApi` does not seem to work for me

Reply:
I've found the app is not actually starting, it completes immediately.
I'm investigating to find the reason.

Reply:
That might be issue with configuration file. I noticed this couple of times, that the match with `Failure` which contains comment //nothing to do, actually does not return anything besides -1.

Reply:
@ajuszczak where exactly?

Reply:
Ok, I think I found the problem here.
The included configuration has no `conseil.security.apiKeys.keys` entry defined.

If I include a value with my custom application.conf, the app starts correctly (even though I suspect logging might not be properly configured, as nothing gets printed when conseil-api starts.
This could explain the fact that no error message was logged when failing to run properly.

Reply:
@ajuszczak do you plan to do any change to the `developer.conf` file, so that the missing keys will be available there?

Also, did you check if the logging config is fine on conseil-api?

Reply:
I can, but as far as I know, this problem was before my changes, so we should treat it more as an improvement. Actually, I think "both" issues are from the past, but that does not change that I'm going to look into it in a second.

Reply:
Thanks for the good work, I guess we can
 * re-think how to present the developer.conf differently and not require it along the import chain, but to tell it's a template to start from
 * simply add the `required(classpath(application))` to developer.conf and suggest the developer to import `developer.conf` as it works right now

Reply:
Regarding logging, there was never such issue in the current  development branch, since the logback file worked for all applications just the same way.
There might be something trivial like a misspelling of a package name or the logback path... I didn't look into that

Reply:
Issue was a bit more complicated. Due to the fact that in build.sbt we had "compile->test" for `common` module, Logback (according to documentation) first picked `Logback-test.xml` file (provided by `common` module due to mentioned operation). To prevent that, I had to get rid of `compile->test` operation, by introducing module as a TestKit. Due to the fact that not all of the classes shared across tests can be generalized and put into `conseil-common-testkit` I decided to temporarily copy them and I'll make it right, once modules will be properly split (specific TODO has been added to it can be identified).